// react-native-form-library/distribution/components/DatePicker/DatePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactNative = require('react-native');

var _DayPicker = require('./DayPicker');

var _DayPicker2 = _interopRequireDefault(_DayPicker);

var _MonthPicker = require('./MonthPicker');

var _MonthPicker2 = _interopRequireDefault(_MonthPicker);

var _YearPicker = require('./YearPicker');

var _YearPicker2 = _interopRequireDefault(_YearPicker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var styles = _reactNative.StyleSheet.create({
  pickerContainer: {
    flexDirection: 'row'
  },
  container: {
    flex: 1
  }
});

var DatePicker = function (_React$Component) {
  _inherits(DatePicker, _React$Component);

  function DatePicker(props) {
    _classCallCheck(this, DatePicker);

    var _this = _possibleConstructorReturn(this, (DatePicker.__proto__ || Object.getPrototypeOf(DatePicker)).call(this, props));

    _this.state = _this.initializeState();
    return _this;
  }

  /*
  * Initialize the state for the month year picker.
  */


  _createClass(DatePicker, [{
    key: 'initializeState',
    value: function initializeState() {
      var date = new Date();
      var defaultValue = this.props.value ? this.props.value.split("-") : null;

      var year = date.getFullYear();
      var month = date.getMonth() + 1;
      var day = date.getDate();

      if (defaultValue) {
        year = defaultValue ? parseInt(defaultValue[0]) : year, month = defaultValue && defaultValue[1] ? parseInt(defaultValue[1]) : month, day = defaultValue && defaultValue[2] ? parseInt(defaultValue[2]) : day;
      }

      this.props.onChange(this.props.name, year + '-' + month + '-' + day);

      return { year: year, month: month, day: day };
    }
    /*
     * Custom on blur function.
     */

  }, {
    key: 'onBlur',
    value: function onBlur() {
      if (this.props.onBlur) this.props.onBlur();
    }

    /*
    * Change the state of the day.
    *
    * @param [String] name. Name passed from the day picker.
    * @param [any] value. Value of the day.
    */

  }, {
    key: 'onDayChange',
    value: function onDayChange(name, value) {
      this.setState({ day: value });
      this.props.onChange(this.props.name, this.state.year + '-' + this.state.month + '-' + value);
    }

    /*
    * Change the state of the month.
    *
    * @param [String] name. Name passed from the month picker.
    * @param [any] value. Value of the month.
    */

  }, {
    key: 'onMonthChange',
    value: function onMonthChange(name, value) {
      this.setState({ month: value });
      this.props.onChange(this.props.name, this.state.year + '-' + value + '-' + this.state.day);
    }

    /*
    * Change the state of the year.
    *
    * @param [String] name. Name passed from the year picker.
    * @param [any] value. Value of the year.
    */

  }, {
    key: 'onYearChange',
    value: function onYearChange(name, value) {
      this.setState({ year: value });
      this.props.onChange(this.props.name, value + '-' + this.state.month + '-' + this.state.day);
    }

    /*
    * Get the days of a month for a given year.
    *
    * @param [int] month. Month, where jan starts at 1.
    * @param [int] year. Year to get days for month.
    */

  }, {
    key: 'getDaysInMonth',
    value: function getDaysInMonth(month, year) {
      //Day 0 is the last day in the previous month
      return new Date(year, month, 0).getDate();
    }
  }, {
    key: 'render',
    value: function render() {
      var picker = this.props;
      var maxDays = this.getDaysInMonth(this.state.month, this.state.year);
      // If month changes and days in month are less than current value of day, reset day
      if (this.state.day > maxDays) {
        this.onDayChange('day', 1);
      }

      return _react2.default.createElement(
        _reactNative.View,
        { style: styles.pickerContainer },
        _react2.default.createElement(
          _reactNative.View,
          { style: styles.container },
          _react2.default.createElement(_MonthPicker2.default, {
            style: picker.style,
            name: 'month',
            onChange: this.onMonthChange.bind(this),
            value: this.state.month
          })
        ),
        _react2.default.createElement(
          _reactNative.View,
          { style: styles.container },
          _react2.default.createElement(_DayPicker2.default, {
            style: picker.style,
            maxDays: maxDays,
            name: 'day',
            onChange: this.onDayChange.bind(this),
            value: this.state.day
          })
        ),
        _react2.default.createElement(
          _reactNative.View,
          { style: styles.container },
          _react2.default.createElement(_YearPicker2.default, {
            style: picker.style,
            name: 'year',
            maxYear: picker.maxYear,
            minYear: picker.minYear,
            onChange: this.onYearChange.bind(this),
            value: this.state.year
          })
        )
      );
    }
  }]);

  return DatePicker;
}(_react2.default.Component);

DatePicker.propTypes = {
  maxYear: _propTypes2.default.number,
  minYear: _propTypes2.default.number,
  name: _propTypes2.default.string,
  onChange: _propTypes2.default.func,
  style: _propTypes2.default.oneOfType([_propTypes2.default.object, _propTypes2.default.array, _propTypes2.default.number]),
  value: _propTypes2.default.string
};

exports.default = DatePicker;
